pkg/operator/rules.go

```diff
@@ -63,11 +64,31 @@ func NewPrometheusRuleSelector(ruleFormat RuleConfigurationFormat, labelSelector
 	}, nil
 }
 
-func generateRulesConfiguration(ruleformat RuleConfigurationFormat, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
-	if ruleformat == PrometheusFormat {
-		// Unset partialResponseStrategy field.
-		for i := range promRule.Groups {
+func generateRulesConfiguration(ruleformat RuleConfigurationFormat, version semver.Version, namespace string, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
+	minVersionLimitsProm := semver.MustParse("2.31.0")
+	minVersionLimitsThanos := semver.MustParse("0.24.0")
+
+	for i := range promRule.Groups {
+		if ruleformat == PrometheusFormat {
+			if promRule.Groups[i].Limit != 0 {
+				if version.LT(minVersionLimitsProm) {
+					level.Warn(logger).Log("msg", fmt.Sprintf("limit` is supported only from Prometheus version %q", minVersionLimitsProm),
+						"prometheusrule", promRule.Groups[i].Name,
+						"namespace", namespace,
+					)
+				}
+			}
+			// Unset partialResponseStrategy field.
```

Comment:
This way the `for` loop to me seems simpler, we can then get rid of the else below 
```suggestion
        minVersionLimits := semver.MustParse("2.31.0")
        component := "Prometheus"
        if ruleformat != PrometheusFormat {
           minVersionLimits := semver.MustParse("0.24.0")
           component := "Thanos"
        } 

	for i := range promRule.Groups {
		if promRule.Groups[i].Limit != 0 {
			if version.LT(minVersionLimits) {
				level.Warn(logger).Log("msg", fmt.Sprintf("`limit` is supported only from %s version %q", component, minVersionLimits),
						"prometheusrule", promRule.Groups[i].Name,
						"namespace", namespace,
					)
				}
			}
		if component == "Prometheus" {
			// Unset partialResponseStrategy field.
```

Reply:
That looks simplified, thanks for the suggestion :)